docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md

```diff
@@ -0,0 +1,169 @@
+
+# Improved error reporting
+Down the error reporting lane, match type reduction errors were improved. When using a match type, it may or may not reduce to one of its cases. If it doesn't match type is used as specified, e.g. if `M[T]` is a match type and it didn't reduce for `M[Int]`, `M[Int]` will be used. This behavior, however, is frequently not what you want: there is a lot of cases where you would expect a match type to reduce but it doesn't. In such cases, it would be nice to have some diagnostic regarding why it didn't reduce. PR [#12053](https://github.com/lampepfl/dotty/pull/12053/) adds just such a diagnostic. E.g. the following code:
+
+```scala
+trait A
+trait B
+type M[X] = X match
+  case A => Int
+  case B => String
+val x: String = ??? : M[B] // error
+```
+
+will report the following error:
+
+```
+6 |val x: String = ??? : M[B] // error
+  |                ^^^^^^^^^^
+  |                Found:    M[B]
+  |                Required: String
+  |
+  |                Note: a match type could not be fully reduced:
+  |
+  |                  trying to reduce  M[B]
+  |                  failed since selector  B
+  |                  does not match  case A => Int
+  |                  and cannot be shown to be disjoint from it either.
+  |                  Therefore, reduction cannot advance to the remaining case
+  |
+  |                    case B => String
+```
+
+# Scaladoc
+We have updated the [documentation](http://dotty.epfl.ch/docs/usage/scaladoc/index.html) for Scaladoc making it easier for you to get started. Also, PR [#11582](https://github.com/lampepfl/dotty/pull/11582) has added the snippet compiler to ensure the snippets in your scaladoc documentation comments aren't broken. You can read more about this feature on the [mailing list](https://contributors.scala-lang.org/t/snippet-validation-in-scaladoc-for-scala-3/4976).
+
+# Metaprogramming
+A lot of metaprogramming work was focused on improving the performance. Some of the notable PRs include:
+
+- Cache quote unpickling [#12242](https://github.com/lampepfl/dotty/pull/12242)
+- Avoid pickled tasty for some captured quote reference [#12248](https://github.com/lampepfl/dotty/pull/12248)
+- Improve quote matcher performance [#12418](https://github.com/lampepfl/dotty/pull/12418)
+- Port scala.quoted.runtime.impl.QuoteMatcher [#12402](https://github.com/lampepfl/dotty/pull/12402)
+
+A way to abort macro expansion was enhanced in PR [#12056](https://github.com/lampepfl/dotty/pull/12056). Now you can use `errorAndAbort(msg: String)` to stop macro expansion with an error, and `expr.valueOrAbort` to abort macro expansion if a given expression doesn't represent a value or possibly contains side effects.
```

Comment:
```suggestion
```

This is experimental. We should not mention it here as users will no be able to use it.